### PR TITLE
Support for table primary keys in SQL to DDlog compiler

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
Fixes #807

File: sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java [hunk 1]
```diff
@@ -73,15 +73,25 @@ protected DDlogIRNode visitCreateTable(CreateTable node, TranslationContext cont
...
+                if (Linq.any(cd.getProperties(), p -> p.getName().getValue().equals("primary_key")))
+                    keyColumns.add(field);
+            }
         }
         DDlogTUser tuser = context.createStruct(node, fields, name);
         String relName = DDlogRelationDeclaration.relationName(name);
         context.reserveGlobalName(relName);
         DDlogRelationDeclaration rel = new DDlogRelationDeclaration(
                 node, DDlogRelationDeclaration.Role.Input, relName, tuser);
+        if (keyColumns.size() > 0)
```
Comment (lalithsuresh) [inline sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java:92]: Use braces just to be sure?

File: sql/src/test/java/ddlog/WeaveTest.java [hunk 2]
```diff
@@ -12,7 +12,7 @@ public void testWeave() {
         Translator t = new Translator(null);
         String node_info = "create table node_info\n" +
                 "(\n" +
-                "  name varchar(36) not null /*primary key*/,\n" +
+                "  name varchar(36) not null with (primary_key = true),\n" +
```
Comment (lalithsuresh) [inline sql/src/test/java/ddlog/WeaveTest.java:15]: Can you also test for multiple primary keys in the same table (compound key)?

Comment (mbudiu-vmw) [reply sql/src/test/java/ddlog/WeaveTest.java:15]: See keyTest

**Comment by lalithsuresh** [comment]
LGTM

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**